src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,28 @@
 package mate.academy;
 
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final ConcurrentHashMap<EventListener, Boolean> listeners = new ConcurrentHashMap<>();
+    private final ExecutorService executorService = Executors.newCachedThreadPool();
```

Comment:
While using Executors.newCachedThreadPool() can be suitable for tasks with a light and bursty load, it can potentially create an unbounded number of threads. If the event rate is high, this could lead to resource exhaustion. Consider using Executors.newFixedThreadPool() with a maximum thread count or implementing a custom ThreadPoolExecutor for more control over the thread pool characteristics.